isthmus/src/test/java/io/substrait/isthmus/integration/ExtendedExpressionIntegrationTest.java

```diff
@@ -0,0 +1,162 @@
+    } catch (Exception e) {
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void projectDataset() throws SqlParseException, IOException {
+    URL resource = ClassLoaderUtil.getClassLoader().getResource("./tpch/data/nation.parquet");
+    String sqlExpression = "20 + N_NATIONKEY";
+    ScanOptions options =
+        new ScanOptions.Builder(/*batchSize*/ 32768)
+            .columns(Optional.empty())
+            .substraitProjection(getProjectExtendedExpression(sqlExpression))
+            .build();
+    try (BufferAllocator allocator = new RootAllocator();
+        DatasetFactory datasetFactory =
+            new FileSystemDatasetFactory(
+                allocator,
+                NativeMemoryPool.getDefault(),
+                FileFormat.PARQUET,
+                resource.toURI().toString());
+        Dataset dataset = datasetFactory.finish();
+        Scanner scanner = dataset.newScan(options);
+        ArrowReader reader = scanner.scanBatches()) {
+      int count = 0;
+      int sum = 0;
+      while (reader.loadNextBatch()) {
+        count += reader.getVectorSchemaRoot().getRowCount();
+        IntVector intVector = (IntVector) reader.getVectorSchemaRoot().getVector(0);
+        for (int i = 0; i < intVector.getValueCount(); i++) {
+          sum += intVector.get(i);
+        }
+        System.out.println(reader.getVectorSchemaRoot().contentToTSVString());
+      }
+      assertEquals(25, count);
+      assertEquals(24 * 25 / 2 + 20 * count, sum);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void filterDatasetUsingExtendedExpression() throws SqlParseException, IOException {
+    URL resource = ClassLoaderUtil.getClassLoader().getResource("./tpch/data/nation.parquet");
+    String sqlExpression = "N_NATIONKEY > 20";
+    ScanOptions options =
+        new ScanOptions.Builder(/*batchSize*/ 32768)
+            .columns(Optional.empty())
+            .substraitFilter(getFilterExtendedExpression(sqlExpression))
+            .build();
+    try (BufferAllocator allocator = new RootAllocator();
+        DatasetFactory datasetFactory =
+            new FileSystemDatasetFactory(
+                allocator,
+                NativeMemoryPool.getDefault(),
+                FileFormat.PARQUET,
+                resource.toURI().toString());
+        Dataset dataset = datasetFactory.finish();
+        Scanner scanner = dataset.newScan(options);
+        ArrowReader reader = scanner.scanBatches()) {
+      int count = 0;
+      while (reader.loadNextBatch()) {
+        count += reader.getVectorSchemaRoot().getRowCount();
+        System.out.println(reader.getVectorSchemaRoot().contentToTSVString());
+      }
+      assertEquals(4, count);
+    } catch (Exception e) {
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static ByteBuffer getFilterExtendedExpression(String sqlExpression)
```

Comment:
Do we need separate methods like `getFilterExtendedExpression` and `getProjectExtendedExpression` ? 
Isn't the functionality same in both functions? Please correct me if I am wrong.

Reply:
Thanks, now just maintain one method.